main.js

```diff
@@ -0,0 +1,124 @@
+// variables
+let saveArr = [];
+let chosenColor = ""
+let boxSize = {x: 30, y: 30};
+let flexBox = document.querySelector("#flex_box");
+let colors = ["red", "blue", "green", "yellow", "white"]
+let palette = document.querySelector(".palette");
+let saveButton = document.querySelector("#save");
+let loadButton = document.querySelector("#load");
+let clearButton = document.querySelector("#clear");
+let colorPicker = document.createElement("input");
+colorPicker.type = "color";
+colorPicker.id = "color_picker"
+colorPicker.value = "#269acb";
+
+// event listeners
+flexBox.addEventListener("click", paint);
+palette.addEventListener("click", colorChoice);
+colorPicker.addEventListener("change", chooser);
+saveButton.addEventListener("click", save);
+loadButton.addEventListener("click", load);
+clearButton.addEventListener("click", clear);
+
+// builds the pixel paint box, depending on the boxSize data
+// rows
+function boxBuilder() {
+  for (let i = 0; i < boxSize.y; i++) {
+    let flexRow = document.createElement("div");
+    flexRow.setAttribute("class","flex_row");
+    // fill the rows & append
+    for (let j = 0; j < boxSize.x; j++) {
+      let pixel = document.createElement("div");
+      pixel.setAttribute("class","pixel");
+      pixel.id = i + "_" + j;
+      pixel.addEventListener("mouseover", brushFx);
+      pixel.addEventListener("mouseleave", fixTheBrush)
+      flexRow.appendChild(pixel);
+    }
+    flexBox.appendChild(flexRow);
+  }
+}
+
+// creates the palette
+palette.appendChild(colorPicker);
```

Comment:
This next bit of code is in between functions. Either group it with other immediately invoke code, or put it in a function.